build.sbt

```diff
@@ -130,19 +128,49 @@ lazy val commonSettings = Seq(
 // General Settings
 inThisBuild(List(
   organization := "org.typelevel",
-  organizationName := "Typelevel Contributors",
-  developers := List(
-    Developer("rossabaker", "Ross A. Baker", "ross@rossabaker.com", url("https://github.com/rossabaker"))
-  ),
+  organizationName := "Typelevel",
+  publishGithubUser := "rossabaker",
+  publishFullName := "Ross A. Baker",
+  baseVersion := "0.3",
 
   homepage := Some(url("https://github.com/typelevel/case-insensitive")),
   startYear := Some(2020),
   licenses := Seq("Apache-2.0" -> url("https://www.apache.org/licenses/LICENSE-2.0.html")),
+  scmInfo := Some(ScmInfo(url("https://github.com/typelevel/case-insensitive"),
+    "git@github.com:typelevel/case-insensitive.git")),
 
-  pomIncludeRepository := { _ => false},
-  scalacOptions in (Compile, doc) ++= Seq(
-      "-groups",
-      "-sourcepath", (baseDirectory in LocalRootProject).value.getAbsolutePath,
-      "-doc-source-url", "https://github.com/typelevel/case-insensitive/blob/v" + version.value + "€{FILE_PATH}.scala"
+  githubWorkflowTargetTags ++= Seq("v*"),
+  githubWorkflowPublishTargetBranches := Seq(RefPredicate.StartsWith(Ref.Tag("v"))),
+  githubWorkflowBuildPreamble := Seq(
+    WorkflowStep.Use("olafurpg", "setup-gpg", "v3"),
+    WorkflowStep.Use("actions", "setup-ruby", "v1"),
+    WorkflowStep.Run(List("""
+      |gem install bundler
+      |bundle install --gemfile=site/Gemfile
+    """.stripMargin), name = Some("Install Jekyll"))
+  ),
+  githubWorkflowBuild := Seq(
+    WorkflowStep.Sbt(List(
+      "headerCheck",
+      "test:headerCheck",
+      "scalafmtCheckAll",
+      "testIfRelevant",
+      "mimaReportBinaryIssuesIfRelevant",
+      "doc",
+      "makeMicrosite"
+    )),
+  ),
+  githubWorkflowPublish := Seq(
+    WorkflowStep.Sbt(List("ci-release")),
+    WorkflowStep.Run(List(s"""
+      |eval "$$(ssh-agent -s)"
+      |echo "$$SSH_PRIVATE_KEY" | ssh-add -
+      |git config --global user.name "GitHub Actions CI"
+      |git config --global user.email "ghactions@invalid"
+      |sbt ++$scala_2_12 site/publishMicrosite
```

Comment:
Fwiw, `WorkflowStep.Sbt(List("site/publishMicrosite"))` will take care of the `++` shenanigans for you, as well as future-proof the build definition a bit if `sbt` ever moves, though it's relatively low-value.